File: src/common/py/features/data_management_plan_feature.py
from .project_feature import ProjectFeature


class DataManagementPlanFeature(ProjectFeature):
    FEATURE_ID = "dmp"

    def __init__(self):
        super().__init__(
            feature_id=DataManagementPlanFeature.FEATURE_ID,
            flags=ProjectFeature.FeatureFlags.OPTIONAL,
            display_name="Data Management Plan",
        )

File: src/common/web/features/DataManagementPlanFeature.ts
import { type ProjectFeatureID } from "../data/entities/EntityTypes";
import { ProjectFeature, ProjectFeatureFlags } from "./ProjectFeature";
import { type ProjectFeaturePanelLoader } from "./ProjectFeaturePanelLoader";

export class DataManagementPlanFeature extends ProjectFeature {
    public static readonly FeatureID: ProjectFeatureID = "dmp";

    public constructor(panel?: ProjectFeaturePanelLoader) {
        super(DataManagementPlanFeature.FeatureID, {
            flags: ProjectFeatureFlags.Optional,
            displayName: "Data Management Plan",
            optionName: "Data Management Plan",
            panel: panel
        });
    }
}
